crowbar_framework/app/models/deployer_service.rb

```diff
@@ -99,9 +101,11 @@ def transition(inst, name, state)
         break if done
       end
 
-      os_map = role.default_attributes["deployer"]["os_map"]
-      node.crowbar["crowbar"]["hardware"]["os"] = os_map[0]["install_os"]
-      node.save
+      unless node.crowbar["crowbar"]["hardware"].nil?
```

Comment:
do you have a link to the original error?

and not sure but how can you assume that `role.default_attributes["deployer"]["os_map"]` is not nil when node.crowbar["crowbar"]["hardware"] is not nil?